bundle-httpcache/src/main/java/com/adobe/acs/commons/httpcache/store/mem/impl/MemHttpCacheStoreImpl.java

```diff
@@ -160,9 +185,15 @@ public void invalidate(HttpCacheConfig cacheConfig) {
         ConcurrentMap<CacheKey, MemCachePersistenceObject> cacheAsMap = cache.asMap();
         for (CacheKey key : cacheAsMap.keySet()) {
             // Match the cache key with cache config.
-            if (cacheConfig.knows(key)) {
-                // If matches, invalidate that particular key.
-                cache.invalidate(key);
+            try {
+                if (cacheConfig.knows(key)) {
+                    // If matches, invalidate that particular key.
+                    cache.invalidate(key);
+                }
+            } catch (HttpCacheKeyCreationException e) {
+                // TODO Verify full invalidation is the best approach in the event of a failure
+                log.error("Could not invalidate cache. Falling back to full cache invalidation.", e);
```

Comment:
@jasiwalgaurav @Sivaramvt WDYT about this? This exception wasnt being handled so I did this .. im cool w whatever we do w this though; invalidating everything seemed like the safest bet.